components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,442 @@
+		ipAddress:   ip,
+		status:      status,
+		Opensearch:  osStatus,
+	}
+}
+
+func (ss *Summary) opensearchStatusInFE(sshUtil SSHUtil) string {
+	arg := []string{
+		"-s",
+		clusterOSHealth,
+	}
+	curlOSStatusScript := GenerateOriginalAutomateCLICommand(
+		&cobra.Command{
+			Use: "curl",
+		}, arg)
+	osStatusJson, _ := sshUtil.connectAndExecuteCommandOnRemote(curlOSStatusScript, true)
+
+	body := []byte(osStatusJson)
+	esHealthData := make(map[string]json.RawMessage)
+
+	err := json.Unmarshal(body, &esHealthData)
+	if err != nil {
+		return "Unknown"
+	}
+
+	return fmt.Sprintf("%s (Active: %s)", esHealthData["status"], esHealthData["active_shards_percent_as_number"])
+}
+
+func executeCmd(sshUtil SSHUtil, cmd string, args []string) (map[string]interface{}, error) {
+	script := GenerateOriginalAutomateCLICommand(&cobra.Command{
+		Use: "curl",
+	}, args)
+	output, err := sshUtil.connectAndExecuteCommandOnRemote(script, true)
+	if err != nil {
+		return nil, err
+	}
+
+	var body map[string]interface{}
+	err = json.Unmarshal([]byte(output), &body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+func (ss *Summary) checkIPAddresses() ([]string, []string, []string, []string, *list.List) {
+	errorList := list.New()
+	var automateIps, chefServerIps, opensearchIps, postgresqlIps []string
+	if ss.statusSummaryCmdFlags.node != "" {
+		automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList = ss.getIpAddressesFromFlag(errorList)
+	} else {
+		if ss.statusSummaryCmdFlags.isAutomate {
+			automateIps = ss.infra.Outputs.AutomatePrivateIps.Value
+		}
+		if ss.statusSummaryCmdFlags.isChefServer {
+			chefServerIps = ss.infra.Outputs.ChefServerPrivateIps.Value
+		}
+		if ss.statusSummaryCmdFlags.isOpenSearch {
+			opensearchIps = ss.infra.Outputs.OpensearchPrivateIps.Value
+		}
+		if ss.statusSummaryCmdFlags.isPostgresql {
+			postgresqlIps = ss.infra.Outputs.PostgresqlPrivateIps.Value
+		}
+		if !ss.statusSummaryCmdFlags.isAutomate &&
+			!ss.statusSummaryCmdFlags.isChefServer &&
+			!ss.statusSummaryCmdFlags.isOpenSearch &&
+			!ss.statusSummaryCmdFlags.isPostgresql {
+			automateIps = ss.infra.Outputs.AutomatePrivateIps.Value
+			chefServerIps = ss.infra.Outputs.ChefServerPrivateIps.Value
+			opensearchIps = ss.infra.Outputs.OpensearchPrivateIps.Value
+			postgresqlIps = ss.infra.Outputs.PostgresqlPrivateIps.Value
+		}
+	}
+	return automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList
+}
+
+func (ss *Summary) getIpAddressesFromFlag(errorList *list.List) ([]string, []string, []string, []string, *list.List) {
+	var automateIps, chefServerIps, opensearchIps, postgresqlIps, nodes []string
+	if ss.statusSummaryCmdFlags.isAutomate ||
+		ss.statusSummaryCmdFlags.isChefServer ||
+		ss.statusSummaryCmdFlags.isOpenSearch ||
+		ss.statusSummaryCmdFlags.isPostgresql {
+		nodes = ss.splitIP(
+			ss.statusSummaryCmdFlags.node,
+		)
+		if len(nodes) != 0 {
+			if ss.statusSummaryCmdFlags.isAutomate {
```

Comment:
Can create a common function like :-
`func checkGivenNode(givenNodes []string, nodeIps []string, service string, errorList *list.List) ([]string, []string, *list.List) {
	ipsFound, ipsNotFound, err := validateIPAddresses(givenNodes, nodeIps, "Automate")
	givenNodes = ipsNotFound
	if err != nil && err.Len() > 0 {
		errorList.PushBack(fmt.Sprintf(getSingleErrorFromList(err).Error(), ipAddressError))
	}
	return ipsFound, givenNodes, errorList
}
`
And call it for if condition as code is common for nearly all if conditions in this function